account_cash_discount_base/models/account_move.py

```diff
@@ -242,3 +242,15 @@ def _reverse_move_vals(self, default_values, cancel=True):
             payment_term = partner.property_supplier_payment_term_id
             res["invoice_payment_term_id"] = payment_term.id
         return res
+
+    @api.model
+    def create(self, values):
+        rec = super().create(values)
+        payment_term = rec.invoice_payment_term_id
+        if not payment_term or rec.move_type not in DISCOUNT_ALLOWED_TYPES:
+            return rec
+
+        for field in ("discount_percent", "discount_delay"):
+            if field not in values:
+                rec[field] = payment_term[field]
```

Comment:
This change will trigger the `INSERT` then an `UPDATE` (about DB).
I think it's possible to do it during the `INSERT` (by updating directly the `values` before calling `super` (only if these fields are not set manually of course)).

Reply:
I think it's hard to get the right payment_term before the super. If the payment term isn't in the values it could be a default from the field or ir.default etc...

Reply:
Something like this (please fill it):
```
move_type = values.get('move_type')
payment_term_id = values.get('invoice_payment_term_id')
payment_term = self.env...
for field in ("discount_percent", "discount_delay"):
    if field not in values:
        values.update({field: payment_term[field]})
```

Reply:
Yes and what happens if payment_term_id is None? Which one do you choose? It's not as simple to just assume that we can skip it because default values in Odoo must be respected but these can come from very different places.

Reply:
You can adapt the `if` part.
`if payment_term and field not in values:`

Reply:
My problem here is not that I'm unable to handle a simple None.

It's not possible to handle the case where no payment term is inside of the values dictionary. In that case Odoo fills the payment term using default values from various possible sources.

Following scenario:

1. I create an account.move without given payment_term_id as value
2. Odoo creates an account.move and uses a defined default for the payment term
3. With your ìf` I would skip setting the `discount_*` values.

I agree that normally there is no default value for `invoice_payment_term_id` BUT it's possible to set one via frontend and additional modules hence the module here would skip the copying again.